### PR TITLE
State: do not add sites to state if we cannot manage them

### PR DESCRIPTION
**Pull request by gwwar**
In some situations we may request a site that does not belong to the current user. For example, when we [query for a site given a site fragment](https://github.com/Automattic/wp-calypso/blob/c56872055c3eac4b7b7d8f5900047255e6b47690/client/my-sites/controller.js#L409).

If this happens, let's make sure the reducer throws these values out, since these sites may be added to the sites sidebar. This is confusing since users are not actually able to manage those sites. I've updated the sites reducer to check for the capabilities property, and not update redux state when this is the case.

### Testing Instructions
- No regressions to switching sites, or recent sites
- We still cannot recreate the case in pbg9X-cvt-p2#comment-51123

### CODE REVIEW

File: client/state/sites/reducer.js [hunk 1]
```diff
@@ -73,7 +73,15 @@ export function items( state = null, action ) {
 		case SITE_RECEIVE:
 		case SITES_RECEIVE:
 			// Normalize incoming site(s) to array
-			const sites = action.site ? [ action.site ] : action.sites;
+			const maybeUnmanageableSites = action.site ? [ action.site ] : action.sites;
+			// filter out anything the current user can't manage
+			const sites = maybeUnmanageableSites.filter( site => {
+				return site && site.capabilities;
```
Comment (rodrigoi) [inline client/state/sites/reducer.js]: we can lose the `return` here. Or we can `import { has } from 'lodash/fp` and just:
```js
const sites = maybeUnmanageableSites.filter( has( 'capabilities' ) );
```

File: client/my-sites/controller.js [hunk 2]
```diff
@@ -238,7 +233,7 @@ function onSelectedSiteAvailable( context ) {
 				.slice( 0, 5 )
 				.filter( recentId => {
 					const site = getSite( state, recentId );
-					return site && site.capabilities;
+					return !! site;
```
Comment (rodrigoi) [inline client/my-sites/controller.js]: we can remove the function here, for a simpler `.filter( recentId => ! isNull( getSite( state, recentId ) ) );`

File: client/state/sites/reducer.js (hunk 1 above)
Comment (gwwar) [reply client/state/sites/reducer.js]: Good point that we don't need a variable here. We're okay here with a simple truthy check, but I added the `!!` to note that we are interested in a boolean vs, trying to map over this. Updated this to a single line in b0ebdfb

File: client/my-sites/controller.js (hunk 2 above)
Comment (gwwar) [reply client/my-sites/controller.js]: Good point that we don't need a variable here. We're okay here with a simple truthy check, but I added the `!!` to note that we are interested in a boolean vs, trying to map over this. Shortened this in b0ebdfb

**Comment by gwwar** [comment]
Thanks for the review @rodrigoi !

**Pull request merged**